### DISCUSSION TITLE
METplus in a Windows OS-based ArcGIS Pro python environment?

### DISCUSSION THREAD

**Discussion by fortworthjack** [dtcenter/METplus#1331] [Installation] [answered]
As I progress through the course, making use of the AWS environment provided for the course, and unsure if it , or another, would be available after the course, I've been wondering if it might be possible to add METplus and its components to a rather rich python environment that comes with ArcGIS Pro.  ArcGIS is Windows-based.  Anyone know if it might be too heavy a lift to try?  If it is possible, might there be any instructions out there.  

Below is a screenshot of some of the packages already available, and what resulted as I used conda to add dateutil, which I saw an exercise or two back. 

Don't get me wrong, I'm fine/happy taking the course in the provided environment, but at some point I might want to integrate some of the METplus workflows with others I'm more familiar with, hence my wondering.   

![image](https://user-images.githubusercontent.com/6979158/146974795-16019225-2560-41a8-85f4-5978eece1322.png)

**Comment by TaraJensen** [comment, marked as answer]
Jack,

I see you are interested in installation options in a Windows/ArcGIS framework.  We do not have experience with installing METplus in that type of environment and I am skeptical it will work.  For Windows machines, we typically recommend using METplus in a container, as it better emulates the Linux environment METplus was designed to work in.  

For more information on how to install METplus via containers, please see the DTC NWP Containers On-line tutorial at: https://dtcenter.org/community-code/numerical-weather-prediction-nwp-containers

METplus and MET containers are available on DockerHub.  Please see our Existing Builds page for more information: https://dtcenter.org/community-code/metplus/metplus-4-0-existing-builds

Let us know if you need further help.

**Reply by fortworthjack** [reply]
Thank you. I'll read up more on the Docker/Container approach, and pursue those.

**Reply by georgemccabe** [reply]
Jack, I'd like to follow up to note that while utilizing Docker has been the easiest approach I have taken to use METplus on a Windows machine, I have also successfully installed METplus in a WSL (Windows Subsystem for Linux) environment.

**Reply by fortworthjack** [reply]
Interesting, thanks George.  I believe I previously set up WSL on my machine, to handle other go-between processes, so if there are any rough instructions out there on how to accomplish an install, I'd be interested in taking a gander.

**Reply by georgemccabe** [reply]
Hi Jack, I followed the instructions to install on a Linux machine to install in WSL. Those instructions can be found under the "Sample Script for Compiling External Libraries and MET" heading of the [MET Download Page](https://dtcenter.org/community-code/model-evaluation-tools-met/download). I don't recall any differences in the installation process specific to WSL.

Since you have marked this discussion topic as answered, I'd like to lock this discussion to prevent any additional comments. We do this to prevent new questions asked in a topic that is marked as answered from being missed by our support staff. If you have any other questions, please feel free to create a new discussion topic or search the existing topics for relevant information.
